common/src/deposits.rs

```diff
@@ -83,11 +116,79 @@ impl DepositScript {
     }
 }
 
+/// Drops the top stack item
+pub const OP_DROP: u8 = opcodes::OP_DROP.to_u8();
+/// <https://en.bitcoin.it/wiki/OP_CHECKSIG> pushing 1/0 for
+/// success/failure.
+pub const OP_CHECKSIG: u8 = opcodes::OP_CHECKSIG.to_u8();
+/// Read the next byte as N; push the next N bytes as an array onto the
+/// stack.
+pub const OP_PUSHDATA1: u8 = opcodes::OP_PUSHDATA1.to_u8();
+
+/// This function checks that the deposit script is valid. Specifically, it
+/// checks that it follows the format laid out in
+/// https://github.com/stacks-network/sbtc/issues/30, where the script is
+/// expected to be
+/// ```text
+///  <deposit-data> OP_DROP OP_PUSHBYTES_32 <x-only-public-key> OP_CHECKSIG
+/// ```
+pub fn parse_deposit_script(deposit_script: &ScriptBuf) -> Result<DepositScriptInputs, Error> {
+    let script = deposit_script.as_bytes();
+
+    // Valid deposit scripts cannot be less than this length.
+    if script.len() < STANDARD_SCRIPT_LENGTH {
+        return Err(Error::InvalidDepositScript);
+    }
+    // This cannot panic because of the above check and the fact that
+    // DEPOSIT_SCRIPT_FIXED_LENGTH < STANDARD_SCRIPT_LENGTH.
```

Comment:
Oh I didn't realize `split_at` was a panicking function. Thanks for the clarification!